### PR TITLE
Update to DOT instead of DOP for OAuth2 Authentication

### PR DESCRIPTION
**Pull request by nasthagiri**
This includes removing dependency on Open ID Connect.

Make sure that the following steps are done before rebasing/merging

  - [ ] Request a review if desired
  - [ ] Squash/Fixup your branch to one commit
  - [ ] Make sure you have updated [edx/configuration](https://github.com/edx/configuration) and [edx/devstack](https://github.com/edx/devstack) if necessary

### CODE REVIEW

File: credentials/settings/devstack.py [hunk 1]
```diff
@@ -45,20 +45,19 @@
...
-SOCIAL_AUTH_EDX_OIDC_KEY = os.environ.get('SOCIAL_AUTH_EDX_OIDC_KEY', 'credentials-key')
-SOCIAL_AUTH_EDX_OIDC_SECRET = os.environ.get('SOCIAL_AUTH_EDX_OIDC_SECRET', 'credentials-secret')
-SOCIAL_AUTH_EDX_OIDC_ISSUER = os.environ.get('SOCIAL_AUTH_EDX_OIDC_ISSUER', 'http://edx.devstack.lms:18000/oauth2')
-SOCIAL_AUTH_EDX_OIDC_URL_ROOT = os.environ.get('SOCIAL_AUTH_EDX_OIDC_URL_ROOT', 'http://edx.devstack.lms:18000/oauth2')
-SOCIAL_AUTH_EDX_OIDC_LOGOUT_URL = os.environ.get('SOCIAL_AUTH_EDX_OIDC_LOGOUT_URL', 'http://localhost:18000/logout')
-SOCIAL_AUTH_EDX_OIDC_PUBLIC_URL_ROOT = os.environ.get('SOCIAL_AUTH_EDX_OIDC_PUBLIC_URL_ROOT',
-                                                      'http://localhost:18000/oauth2')
-SOCIAL_AUTH_EDX_OIDC_ID_TOKEN_DECRYPTION_KEY = os.environ.get('SOCIAL_AUTH_EDX_OIDC_ID_TOKEN_DECRYPTION_KEY',
-                                                              'credentials-secret')
+OAUTH2_PROVIDER_URL = 'http://localhost:18000/oauth2'
```
Comment (doctoryes) [inline credentials/settings/devstack.py]: Did this OAuth2 handshake *never* work because of this naming? Or is this change needed because of the way the new backend works?

File: credentials/settings/local.py [hunk 2]
```diff
@@ -39,10 +39,9 @@
 
 # Set these to the correct values for your OAuth2/OpenID Connect provider (e.g., devstack)
```
Comment (doctoryes) [inline credentials/settings/local.py]: Missed an "OpenID" above...

File: credentials/apps/core/models.py [hunk 3]
```diff
@@ -135,15 +135,15 @@ def __str__(self):
...
 
     @property
     def oauth2_client_id(self):
-        return settings.SOCIAL_AUTH_EDX_OIDC_KEY
+        return settings.SOCIAL_AUTH_EDX_OAUTH2_KEY
 
     @property
     def oauth2_client_secret(self):
-        return settings.SOCIAL_AUTH_EDX_OIDC_SECRET
+        return settings.SOCIAL_AUTH_EDX_OAUTH2_SECRET
```
Comment (doctoryes) [inline credentials/apps/core/models.py]: Line 250 below defines a custom user model for OpenIDConnect. Maybe mark it deprecated?

File: docs/getting_started.rst [hunk 4]
```diff
@@ -44,44 +44,37 @@ The following example shows how to use the Amazon S3 storage backend::
...
-| SOCIAL_AUTH_EDX_OIDC_SECRET                         | OAuth 2.0 client secret                                                    | (This should be set to the value generated when the client was created.) |
+| SOCIAL_AUTH_EDX_OAUTH2_SECRET                         | OAuth 2.0 client secret                                                    | (This should be set to the value generated when the client was created.) |
 +-----------------------------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------------+
-| SOCIAL_AUTH_EDX_OIDC_URL_ROOT                       | OAuth 2.0 authentication URL                                               | http://127.0.0.1:8000/oauth2                                             |
+| SOCIAL_AUTH_EDX_OAUTH2_URL_ROOT                       | OAuth 2.0 authentication URL                                               | http://127.0.0.1:8000/oauth2                                             |
```
Comment (pwnage101) [inline docs/getting_started.rst]: seems like this should be the LMS port `:18000`
